### ISSUE TITLE
this has a practical application have you tried adding new architecture?

### ISSUE THREAD

**Issue by raoofha** [vojtechkral/asm-polyglot#1]
hi very cool project. 
I just want to say this has a practical application look at [cosmopolitan](https://github.com/jart/cosmopolitan) and [ksuapp.ksuapp](https://github.com/raoofha/ksuapp.ksuapp) using this method we can have a single file to run on all platform (hopefully)

**Comment by vojtechkral** [comment]
Other architectures could be added in a similar way to ARM, with more or less of a difficulty depending on the specific platform.

I think I've read about Cosmopolitan before. There was also some effort to add support for a cross-platform binary format in Linux. But honestly I'm not convinced these project have practical applications either :smile:

**Comment by raoofha** [comment]
so you think trying to make a single file to run everywhere is a wast of effort?

**Comment by vojtechkral** [comment]
@raoofha No, I wouldn't go that far, maybe someone has some good usage for that. For me personally it's fairly easy to just build several binaries & the challenge of cross-platform things is elsewhere to me (APIs, ...)

**Comment by raoofha** [comment]
as an end user it bothers me that linux android ios windows all have their own packaging format although on mobile their all using zip format ( that I hope I can leverage if signing doesn't get in the way ), I just want one file and be able to share offline whenever and wherever I want and anyway for me it's just very cool to have one file it makes understanding of how computer works much much easier, one file and one source of causation, and you can also customize your app/os by a zip or iso file manager that most OSs support very well.
